src/main/java/org/wildfly/security/sasl/util/SortedMechanismSaslServerFactory.java

```diff
@@ -49,7 +50,14 @@ public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final
     public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final String... mechanismNames) {
         super(delegate);
         checkNotNullParam("mechanismNames", mechanismNames);
-        this.saslMechanismSelector = SaslMechanismSelector.NONE.addMechanisms(mechanismNames);
+        this.saslMechanismSelector = () -> SaslMechanismSelector.NONE.addMechanisms(mechanismNames);
```

Comment:
Probably better to start using ``org.wildfly.common.function.Functions#constantSupplier`` instead for these cases.  No time like the present.